modules/standard/Random.chpl

```diff
@@ -1003,6 +1207,43 @@ module Random {
       return res;
     }
 
+    /*
+      Sample ``n`` random elements from an array, where each array element has
+      a corresponding weight.
+
+      Elements with relatively larger weights are more likely to be sampled.
+
+      :arg arr: The 1D rectangular array to sample from. Its domain's ``idxType``
+                should be coercible from this stream's ``eltType``.
+      :arg n: The number of elements to sample
+      :arg weights: An array of weights corresponding to the elements in ``arr``
+      :arg withReplacement: Whether or not to sample with replacement
+
+      :return: A zero-based array of ``n`` random elements sampled from the array
+
+      :throws IllegalArgumentError: If ``n < 1`` or if ``n > arr.size`` and
+                                    ``withReplacement=false``. If ``arr`` is
+                                    empty. If ``weights`` does not have the same
+                                    size as ``arr``.
+    */
+    proc ref sample(const ref arr: [?d] ?t, n: int, const ref weights: [?dw] ?wt, withReplacement=false): [] t throws
```

Comment:
This line looks longer than I would expect, let's try to keep lines to 80 characters as much as we can.  This isn't the only place I noticed it